src/toploop/toploop_jsoo.ml

```diff
@@ -128,3 +128,26 @@ let stop_channel_redirection redir =
         Sys_js.set_channel_flusher redir.channel append ;
   with Not_found ->
     fail ()
+
+let use_compiled_string code =
+  (* jsoo supports dynload, but relies on expectations on the parent object that
+     are no longer valid when running from a web-worker. Thus we compile with
```

Comment:
Why is it no longer valid ? Can anything be done on the jsoo side ? Consider opening a jsoo issue.

Reply:
Ha, I am sorry, that was too long ago for me to remember precisely :grimacing: 
Indeed this could probably be worked around in jsoo; from what I remember, just trying to dynload from within a webworker is enough to point to the issue, with errors about stuff missing from the parent object.